File: 6.1_Graphical User Interface - Calculator.py
# oding = utf-8
# -*- coding:utf-8 -*-
from tkinter import *
# 创建窗口
window = Tk()
# 设置窗口标题
window.title("简易计算器")
# 创建文本框
entry = Entry(window, width=30)
entry.grid(row=0, column=0, columnspan=4)
# 定义按钮点击事件
def button_click(number):
    current = entry.get()
    entry.delete(0, END)
    entry.insert(0, current + str(number))
def button_clear():
    entry.delete(0, END)
def button_calculate():
    expression = entry.get()
    result = eval(expression)
    entry.delete(0, END)
    entry.insert(0, result)
# 创建按钮
button_1 = Button(window, text="1", padx=20, pady=10, command=lambda: button_click(1))
button_2 = Button(window, text="2", padx=20, pady=10, command=lambda: button_click(2))
# 其他按钮的创建类似，省略
# 按钮的布局
button_1.grid(row=1, column=0)
button_2.grid(row=1, column=1)
# 其他按钮的布局类似，省略
# 清除按钮和计算按钮的创建和布局，省略
# 运行窗口循环
window.mainloop()

File: 6.1_Graphical User Interface - Calculator.cpp
#include <gtk/gtk.h>
// 声明窗口和组件
GtkWidget* window;
GtkWidget* button_1;
// 其他按钮的声明，省略
// 按钮点击事件
void button_click(GtkWidget* widget, gpointer data) {
    const gchar* text = gtk_button_get_label(GTK_BUTTON(widget));
    const gchar* current = gtk_entry_get_text(GTK_ENTRY(entry));
    gchar* new_text = g_strconcat(current, text, NULL);
    gtk_entry_set_text(GTK_ENTRY(entry), new_text);
    g_free(new_text);
}
// 清除按钮点击事件
void button_clear(GtkWidget* widget, gpointer data) {
    gtk_entry_set_text(GTK_ENTRY(entry), "");
}
// 计算按钮点击事件
void button_calculate(GtkWidget* widget, gpointer data) {
    const gchar* expression = gtk_entry_get_text(GTK_ENTRY(entry));
    // 进行计算，并将结果显示在文本框中
}
int main(int argc, char* argv[]) {
    // 初始化GTK库
    // 创建窗口
    // 创建文本框
    // 创建按钮
    // 其他按钮的创建和事件绑定，省略
    // 显示窗口和组件
    // 运行GTK主循环
    gtk_main();
}